src/main/java/com/parpiiev/time/annotations/PasswordConstraintValidator.java

```diff
@@ -0,0 +1,58 @@
+package com.parpiiev.time.annotations;
+
+import org.passay.*;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class PasswordConstraintValidator
+        implements ConstraintValidator<ValidPassword, String> {
+
+    @Override
+    public void initialize(ValidPassword arg0) {
+    }
+
+    @Override
+    public boolean isValid(String password, ConstraintValidatorContext context) {
+        PasswordValidator validator = new PasswordValidator(Arrays.asList(
+                // at least 8 characters
+                new LengthRule(1, 8),
+
+                // at least one upper-case character
+//                new CharacterRule(EnglishCharacterData.UpperCase, 1),
+
+                // at least one lower-case character
+                new CharacterRule(EnglishCharacterData.LowerCase, 1),
+
+                // at least one digit character
+                new CharacterRule(EnglishCharacterData.Digit, 1),
+
+                // at least one symbol (special character)
+//                new CharacterRule(EnglishCharacterData.Special, 1),
+
+                // no whitespace
+                new WhitespaceRule()
+
+        ));
+
+        RuleResult result = validator.validate(new PasswordData(password));
+
+        if (result.isValid()) {
+            return true;
+        }
+
+        List<String> messages = validator.getMessages(result);
+
+        String messageTemplate = messages.stream()
```

Comment:
it can be done simpler:
```suggestion
        String messageTemplate = String.join(",", messages);
```